src/test/java/org/prnhs/javaee/swim/core/dao/UserDaoIT.java

```diff
@@ -0,0 +1,104 @@
+package org.prnhs.javaee.swim.core.dao;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import org.hibernate.exception.ConstraintViolationException;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.prnhs.javaee.swim.core.entity.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.AutoConfigureTestDatabase.Replace;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.orm.jpa.JpaSystemException;
+import org.springframework.test.context.junit4.SpringRunner;
+
+
+@RunWith(SpringRunner.class)
```

Comment:
we will leverage spring to run our tests.  Spring will build the necessary scaffolding to test the items specified here, including all necessary object dependencies.